pgjdbc/src/main/java/org/postgresql/jdbc/PgConnection.java

```diff
@@ -1480,12 +1480,31 @@ public void abort(Executor executor) throws SQLException {
     }
   }
 
-  public void setNetworkTimeout(Executor executor, int milliseconds) throws SQLException {
-    throw org.postgresql.Driver.notImplemented(this.getClass(), "setNetworkTimeout(Executor, int)");
+  public void setNetworkTimeout(Executor executor /*not used*/, int milliseconds) throws SQLException {
```

Comment:
This method should check to see that there is an SQLPermission object before allowing the method to proceed. If a SecurityManager exists and its checkPermission method denies calling setNetworkTimeout, this method throws a java.lang.SecurityException.